galloc/docs/optimisation.md

```diff
@@ -0,0 +1,320 @@
+# Optimization of galloc
+We've tried to optimize galloc in some ways, but none of them showed significant performance improvement, so we'd **strongly suggest keeping the current implementation.**
```

Comment:
This must be after `Content` title  in our case. In content this part must be named as `Summary` for developers who does not want to spent their time for detailed investigation.